src/app/modules/main/wallet_section/assets/module.nim

```diff
@@ -123,59 +121,22 @@ proc setAssetsAndBalance(self: Module, tokens: seq[WalletTokenDto]) =
   let totalCurrencyBalanceForAllAssets = tokens.map(t => t.getCurrencyBalance(enabledChainIds, currency)).foldl(a + b, 0.0)
     
   self.view.getAssetsModel().setItems(items)
-  self.view.setCurrencyBalance(currencyAmountToItem(totalCurrencyBalanceForAllAssets, currencyFormat))
 
 method switchAccount*(self: Module, accountIndex: int) =
+  var walletAccount = self.controller.getWalletAccount(accountIndex)
   self.currentAccountIndex = accountIndex
+  if walletAccount.isNil:
```

Comment:
@alaibe it won't get into this if ever, cause `WalletAccountDto` is `object` (won't be `nil` ever cause it's not `ref object`). So it makes more sense to check `if walletAccount.address.len == 0:`

Reply:
This is what we did in order to fix the case where we remove selected account in overview module. This is exact same logic here

Reply:
@alaibe yes I remember that's added few days ago, but looking at it now are you saying that `walletAccount` may be `nil` in some cases? If so, that's very weird. 

From this line `126`: 
`var walletAccount = self.controller.getWalletAccount(accountIndex)`

This proc from wallet service is called:
```
proc getWalletAccount*(self: Service, accountIndex: int): WalletAccountDto =
    let accounts = self.getWalletAccounts()
    if accountIndex < 0 or accountIndex >= accounts.len:
      return
    return accounts[accountIndex]
```
Now even if the index is out or range, returned value is an empty `WalletAccountDto`, so no way to be `nil` ever.

I guess maybe something else was an issue there, but was somehow solved by adding `if walletAccount.isNil:` check.

Reply:
@saledjenic let's imagine this case:
- you have 4 accounts
- current account index is 3
- you delete account at index 3

Then:
- we receive event: account deleted
- we do: self.switchAccount(self.currentAccountIndex)

So current index is 3, we then call getWalletAccount with index 3.
According to the code we enter this if:
`if accountIndex < 0 or accountIndex >= accounts.len:`
which means we return nil

Reply:
@alaibe ahhh I overlooked something, it's declared as `WalletAccountDto* = ref object of RootObj`, not sure now what type I was looking at. 😞 Since it's `ref object` then all good. Sorry for confusion.